calculator.rb

```diff
@@ -0,0 +1,112 @@
+# welcome message
+puts "------ Welcome to my calculator! ------"
+puts "\nLet's do some calculations."
+print "Would you like to add (+), subtract (-), multiply (*), divide (/),
+exponify (**), or find the remainder (%) of two values? > "
+user_input = gets.chomp.downcase
+
+operations = ["add", "+", "subtract", "-", "multiply", "*", "divide", "/",
+"exponify", "**", "remainder", "%"]
+
+# prompt user to select an option from the array
+until operations.include?(user_input)
+    puts "I am sorry, that is not an option"
+    print "Would you like to add (+), subtract (-), multiply (*), divide (/),
+    exponify (**), or find the remainder (%) of two values? > "
+    user_input = gets.chomp.downcase
+end
+
+# prompt user to input first value for the equation
+puts "Great! Now give us two values that you want to use: "
+print "What is your first value? > "
+num1 = gets.chomp
+
+# outputs error until user provides a valid input
+until /^[\d]+?/.match(num1)
+  puts "Please give valid number values"
+  print "What is your first value? > "
+  num1 = gets.chomp
+end
+
+# prompt user to input second value for the equation
+print "What is your second value? > "
+num2 = gets.chomp
+
+# outputs error until user provides a valid input
+until /^[\d]+?/.match(num2)
+  puts "Please give valid number values"
+  print "What is your second value? > "
```

Comment:
You've written almost exactly the same code here as you did above - the only difference is the variable name, `num2` or `num1`. Could you use a method to consolidate these?